network/containerizer/shim.go

```diff
@@ -0,0 +1,129 @@
+// raw state.LinkLayerDevice.
+type linkLayerDevice struct {
+	*state.LinkLayerDevice
+}
+
+// ParentDevice implements LinkLayerDevice by wrapping the response from the
+// inner device's call in a new instance of linkLayerDevice.
+func (l *linkLayerDevice) ParentDevice() (LinkLayerDevice, error) {
+	dev, err := l.LinkLayerDevice.ParentDevice()
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	return &linkLayerDevice{dev}, nil
+}
+
+var _ LinkLayerDevice = (*linkLayerDevice)(nil)
+
+// Machine is an indirection for state.Machine,
+// describing a machine that is to host containers.
+type Machine interface {
+	Id() string
+	AllSpaces() (set.Strings, error)
+	LinkLayerDevicesForSpaces([]string) (map[string][]LinkLayerDevice, error)
+	SetLinkLayerDevices(devicesArgs ...state.LinkLayerDeviceArgs) (err error)
+
+	// TODO (manadart 2018-10-10) These methods are used in tests, which rely
+	// on the StateSuite. Some of them are recruited via the Container
+	// interface below, but they are all located here for simplicity.
+	// A better approach could be sought that does not require their
+	// presence here.
+	SetDevicesAddresses(devicesAddresses ...state.LinkLayerDeviceAddress) (err error)
+	SetParentLinkLayerDevicesBeforeTheirChildren(devicesArgs []state.LinkLayerDeviceArgs) error
+	AllLinkLayerDevices() ([]LinkLayerDevice, error)
+	SetConstraints(cons constraints.Value) (err error)
+	RemoveAllAddresses() error
+	Wrapped() *state.Machine
+}
+
+// MachineShim implements Machine.
+// It is required to mock the return of LinkLayerDevicesForSpaces,
+// which includes raw state.LinkLayerDevice references.
+type MachineShim struct {
+	*state.Machine
+}
+
+// LinkLayerDevicesForSpaces implements Machine by unwrapping the inner
+// state.Machine call and wrapping the raw state.LinkLayerDevice references
+// with the local LinkLayerDevice implementation.
+func (m *MachineShim) LinkLayerDevicesForSpaces(spaces []string) (map[string][]LinkLayerDevice, error) {
+	spaceDevs, err := m.Machine.LinkLayerDevicesForSpaces(spaces)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	wrapped := make(map[string][]LinkLayerDevice, len(spaceDevs))
+	for space, devs := range spaceDevs {
+		wrappedDevs := make([]LinkLayerDevice, len(devs))
+		for i, d := range devs {
+			wrappedDevs[i] = &linkLayerDevice{d}
+		}
+		wrapped[space] = wrappedDevs
+	}
+	return wrapped, nil
+}
+
+// AllLinkLayerDevices implements Machine by wrapping each
+// state.LinkLayerDevice reference in returned collection with the local
+// LinkLayerDevice implementation.
+func (m *MachineShim) AllLinkLayerDevices() ([]LinkLayerDevice, error) {
+	devs, err := m.Machine.AllLinkLayerDevices()
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	wrapped := make([]LinkLayerDevice, len(devs))
+	for i, d := range devs {
+		wrapped[i] = &linkLayerDevice{d}
+	}
+	return wrapped, nil
+}
+
+// Wrapped returns the inner state.Machine reference.
+func (m *MachineShim) Wrapped() *state.Machine {
```

Comment:
Wouldn't this be "Unwrapped" ? or maybe Raw?

Reply:
Happy to go with "Raw".